app/client/src/components/ads/Dropdown.tsx

```diff
@@ -425,8 +444,35 @@ const ErrorLabel = styled.span`
   color: ${Colors.POMEGRANATE2};
 `;
 
+const StyledText = styled(Text)`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
+function TooltipWrappedText(
+  props: TextProps & {
+    label: string;
+  },
+) {
+  const { label, ...textProps } = props;
+  const targetRef = useRef<HTMLDivElement | null>(null);
+  return (
+    <Tooltip
+      boundary="window"
+      content={label}
+      disabled={!isEllipsisActive(targetRef.current)}
+      position={Position.TOP}
+    >
+      <StyledText ref={targetRef} {...textProps}>
+        {label}
+      </StyledText>
```

Comment:
```suggestion
      isEllipsisActive(targetRef.current) ? (
        <StyledText ref={targetRef} {...textProps}>
          {label}
        </StyledText>
      ): <Text type={TextType.P1}>{option.label}</Text>
```
Actually I meant adding ellipsis only if the tooltip is active